src/components/CampaignTextingHoursForm.jsx

```diff
@@ -0,0 +1,234 @@
+import type from 'prop-types'
+import Toggle from 'material-ui/Toggle'
+import React from 'react'
+import Form from 'react-formal'
+import GSForm from './forms/GSForm'
+import CampaignFormSectionHeading from './CampaignFormSectionHeading'
+import yup from 'yup'
+import { css, StyleSheet } from 'aphrodite'
+import cloneDeep from 'lodash/cloneDeep'
+import isEqual from 'lodash/isEqual'
+import moment from 'moment'
+import Autocomplete from 'material-ui/AutoComplete'
+import { dataSourceItem } from './utils'
+
+export default class CampaignTextingHoursForm extends React.Component {
+  state = {
+    showForm: false,
+    timezoneSearchText: undefined,
+    textingHoursStartSearchText: undefined,
+    textingHoursEndSearchText: undefined
+  }
+
+  formSchema = yup.object({
+    overrideOrganizationTextingHours: yup.boolean(),
+    textingHoursEnforced: yup.boolean(),
+    textingHoursStart: yup.number().integer(),
+    textingHoursEnd: yup.number().integer(),
+    timezone: yup.string()
+  })
+
+  fireOnChangeIfTheFormValuesChanged(fieldName, newValue) {
+    const formValues = cloneDeep(this.props.formValues)
+    formValues[fieldName] = newValue
+    if (!isEqual(formValues, this.props.formValues)) {
+      this.props.onChange(formValues)
+    }
+  }
+
+  addToggleFormField(name, label) {
+    return (
+      <Form.Field
+        name={name}
+        type={Toggle}
+        defaultToggled={this.props.formValues[name]}
+        label={label}
+        onToggle={async (_, isToggled) => {
+          this.fireOnChangeIfTheFormValuesChanged(name, isToggled)
+        }}
+      />
+    )
+  }
+
+  addAutocompleteFormField(
+    name,
+    stateName,
+    initialValue,
+    label,
+    hint,
+    choices,
+    collection
```

Comment:
what is this used for? `collection` ?

Reply:
`choices` contains the strings to be displayed

`collection` is the original values, used by a callback created by this function, so when a choice is selected, we can find the value to include when we fire the selected event